client-spark/common/src/main/java/org/apache/spark/shuffle/writer/ShuffleHandleInfoWrapper.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.writer;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.spark.shuffle.ShuffleHandleInfo;
+
+import org.apache.uniffle.common.ShuffleServerInfo;
+import org.apache.uniffle.common.exception.RssException;
+
+/** This class is to wrap the shuffleHandleInfo to speed up the partitionAssignment getting. */
+public class ShuffleHandleInfoWrapper {
```

Comment:
Why not adding this caching to `ShuffleHandleInfo`? With `ShuffleHandleInfo.updateAssignmentPlan` fetching and storing `latestAssignment` and `ShuffleHandleInfo.getPartitionAssignment(taskAttemptId)` providing the assignment from that `Map`.

Reply:
Latest `ShuffleHandleInfo` is stored in spark driver shuffleManager. But for shuffle writer, the holding shuffleHandleInfo is partially latest, which is updated by the grpc handle. 

If using your way, it maybe cause some questions. Because the same object has different usage. It looks not clear.

Feel free to discuss more.

Reply:
You are saying you do not want the `ShuffleHandleInfo` hold by the shuffle writer to change?

Reply:
Not accurate, I hope the updated latest cache is not maintained into the shuffleHandleInfo. Because it is used by the shuffle writer, but the shuffleHandleInfo is transferred by the grpc to writer. So this cache is not good for the shareable for all the tasks. maybe using a independent handle wrapper to hold cache is more clear.

Reply:
Calling `ShuffleHandleInfoWrapper(taskAttemptId, shuffleHandleInfo).retrievePartitionAssignment(partitionId)` could be replaced with `shuffleHandleInfo.getLatestAssignmentPlan(taskAttemptId).get(partitionId)`. To avoid fetching the whole plan for each partition, this wrapper caches it. Looks like this is the only purpose of this class, as indicated by the comment above

    /** This class is to wrap the shuffleHandleInfo to speed up the partitionAssignment getting. */

Why can't `ShuffleHandleInfo` cache the partition assignment? Do you need `ShuffleHandleInfo` to be immutable?

Reply:
> Why can't ShuffleHandleInfo cache the partition assignment? 

Yes. the cache is also OK, let me think twice

> Do you need ShuffleHandleInfo to be immutable?

Needn't. The shuffleHandleInfo will be updated when the reassign occurs. But this will happen in driver side and then send back the latest handleInfo to executor task side.